http_server/parse_start.py

```diff
@@ -0,0 +1,138 @@
+
+        path_stream = stream.Signature(8).create()
+        path_streams = []
+        path_match = []
+        for i,path in enumerate(self._paths):
+            matcher = m.submodules[f"path_matchers_{i}"] = StringContainsMatch(path)
+            path_streams.append(matcher.input)
+            resets.append(matcher.reset)
+            m.d.comb += self.path[i+1].eq(matcher.accepted)
+            path_match.append(matcher.accepted)
+        stream_utils.fanout_stream(m, path_stream, path_streams)
+        any_path_match = stream_utils.tree_or(m, path_match)
+        m.d.comb += self.path[0].eq(~any_path_match)
+
+        # TODO: #4 - If we want to get out of the stone age, should match more than HTTP/1.0
+        #            That being said, silicon is kind of like a stone, right?
+        protocol_match = m.submodules.protocol_match = StringContainsMatch("HTTP/1.0")
+        m.d.comb += self.protocol[self.PROTOCOL_NO_MATCH].eq(~protocol_match.accepted) 
+        m.d.comb += self.protocol[self.PROTOCOL_HTTP1_0].eq(protocol_match.accepted) 
+
+        with m.FSM():
+            with m.State("reset"):
+                for r in resets:
+                    m.d.sync += r.eq(1)
+                m.d.sync += self.done.eq(0)
+                m.next = "match_method"
+            with m.State("match_method"):
+                m.next = "match_method"
+                for r in resets:
+                    m.d.sync += r.eq(0)
+                m.d.comb += [
+                    method_stream.valid.eq(self.input.valid),
+                    method_stream.payload.eq(self.input.payload),
+                    self.input.ready.eq(method_stream.ready),
+                ]
+                with m.If(self.input.valid & (self.input.payload == ord(' '))):
+                    m.next = "match_path"
+            with m.State("match_path"):
+                m.next = "match_path"
+                m.d.comb += [
+                    path_stream.valid.eq(self.input.valid),
+                    path_stream.payload.eq(self.input.payload),
+                    self.input.ready.eq(path_stream.ready),
+                ]
+                with m.If(self.input.valid & (self.input.payload == ord(' '))):
+                    m.next = "match_protocol"
+            with m.State("match_protocol"):
+                m.next = "match_protocol"
+                # connect results in warning about combinatorial signals
+                m.d.comb += [
+                    protocol_match.input.valid.eq(self.input.valid),
+                    protocol_match.input.payload.eq(self.input.payload),
+                    self.input.ready.eq(protocol_match.input.ready),
+                ]
+                with m.If(self.input.valid & (self.input.payload == ord('\r'))):
+                    m.next = "match_end"
+            with m.State("match_end"):
+                m.next = "match_end"
+                with m.If(self.input.valid & (self.input.payload == ord('\n'))):
+                    m.next = "done"

```

Comment:
`m.Else` error?

`GET /hello HTTP/1.0\rPOST /goodbye HTTP/1.1\r\n` should produce an error, not a match for the `POST`.